### PR TITLE
Array

### PR DESCRIPTION
**Pull request by YCGreen**

### CODE REVIEW

**Comment by YCGreen** [comment]
I don't think I set up checkstyles properly -- there's nothing coming up and I know it's not because my code is so stunningly clean. Not sure how to fix that.

**Comment by andrewoid** [comment]
> I don't think I set up checkstyles properly -- there's nothing coming up and I know it's not because my code is so stunningly clean. Not sure how to fix that.

You didn't add all the lines to `.gitignore` and you didn't add `checkstyles.xml` and `java_gradle.yaml`
This is why you have added files in the .idea directory that shouldn't be there.

File: src/test/java/green/fallingsand/SandTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+        //when
+        String actual = sand.toString();
+
+        //then
+        assertEquals("000\n000\n000\n", actual);
+    }
+
+    @Test
+    public void fall() {
```
Comment (andrewoid) [inline src/test/java/green/fallingsand/SandTest.java:34]: Your fall() test is exactly the same as your fallOnGround test. 

That's not how I have it in my PR.

File: src/main/java/green/fallingsand/Sand.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     * Moves all sand down one square
+     */
+    public void fall() {
+        for(int y = 0; y < fieldY; y++) {
+            for(int x = 0; x < fieldX; x++) {
+                if(field[y][x] == 1) {
+                    boolean atBottom;
+                    do {
+                        atBottom = fallSingular(x, y);
+                    } while (!atBottom);
```
Comment (andrewoid) [inline src/main/java/green/fallingsand/Sand.java]: This method is supposed to move all the sand down ONE square. 
Your code doesn't do that.

File: src/main/java/green/fallingsand/Sand.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                }
+            }
+        }
+    }
+
+    public boolean fallSingular(int x, int y) {
+        boolean isBottom = true;
+        if(y < fieldY - 1 && field[y + 1][x] == 0){
+            field[x][y+1] = 1;
+            isBottom = false;
```
Comment (andrewoid) [inline src/main/java/green/fallingsand/Sand.java]: You don't need isBottom. 
Just return false or return true.

But you shouldn't have this method anyway.

File: src/main/java/green/fallingsand/Sand.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
+package green.fallingsand;
+
+public class Sand {
+    private int fieldX = 3;
+    private int fieldY = 3;
```
Comment (andrewoid) [inline src/main/java/green/fallingsand/Sand.java]: these should be constants.

**Comment by andrewoid** [comment]
Your `.github/workflows` is in the wrong place. That's why GitHub isn't running checkstyles or your unit tests.
You've also committed files in the .idea directory that shouldn't be in git.

To remove them run the following command. 

``` bash
git rm --cached -r .idea
git add .
git commit -m "removed .idea directory from git"
```

Then move the `.github/workflows` directory.
Then fix your `.gitignore`

**Pull request merged**